google/cloud/sql/connector/connector.py

```diff
@@ -130,10 +130,22 @@ def __init__(
         # set default params for connections
         self._timeout = timeout
         self._enable_iam_auth = enable_iam_auth
-        self._ip_type = ip_type
         self._quota_project = quota_project
         self._sqladmin_api_endpoint = sqladmin_api_endpoint
         self._user_agent = user_agent
+        # if ip_type is str, convert to enum
+        if isinstance(ip_type, str):
```

Comment:
nit: this and the code below could be wrapped up into a private function to avoid duplication and help long term maintenance.

Reply:
Added the code as a private static method on the `IPTypes` class, this way it is marked as a utility function and tightly coupled with the class.